CodeQL analysis failing - Lombok incompatibility

CodeQL analysis is failing. Snippet from the exception:

```
Caused by: java.lang.ClassNotFoundException: com.sun.tools.javac.code.TypeTags
  [2021-03-28 18:58:03] [autobuild]     at java.lang.ClassLoader.findClass (ClassLoader.java:719)
  [2021-03-28 18:58:03] [autobuild]     at java.lang.ClassLoader.loadClass (ClassLoader.java:589)
  [2021-03-28 18:58:03] [autobuild]     at lombok.launch.ShadowClassLoader.loadClass (ShadowClassLoader.java:422)
```

The CodeQL build has started using Java 11 (up from 8) which is incompatible with the version of Lombok we currently use.

--- a/pom.xml
+++ b/pom.xml
@@ -236,7 +236,7 @@
     <dependency>
       <groupId>org.projectlombok</groupId>
       <artifactId>lombok</artifactId>
-      <version>1.16.16</version>
+      <version>1.18.18</version>
       <scope>provided</scope>
     </dependency>
     <dependency>

